### PR TITLE
fix `@manipulate` for upcoming base changes

### PR DESCRIPTION
**Pull request by simeonschaub**
ref JuliaLang/julia#41857

### CODE REVIEW

File: src/manipulate.jl [hunk 1]
```diff
@@ -79,6 +79,10 @@ macro manipulate(args...)
     block = expr.args[2]
     if expr.args[1].head == :block
         bindings = expr.args[1].args
+        # remove trailing LineNumberNodes from loop body as to not just return `nothing`
```
Comment (piever) [inline src/manipulate.jl]: I'm not 100% sure I understand what will change in newer julia versions, but the loop body is `block` , defined a few lines above. `expr` is a `for` loop, so the body should be its second argument.

File: src/manipulate.jl [hunk 2]
```diff
@@ -79,6 +79,10 @@ macro manipulate(args...)
...
+        if bindings[end] isa LineNumberNode
+            resize!(bindings, length(bindings) - 1)
```
Comment (piever) [inline src/manipulate.jl]: Why not just `pop!` instead of `resize!(v, length(v) - 1)`? I think it'd be a bit more readable.

**Review by piever — CHANGES_REQUESTED** [review]

**Comment by piever** [comment]
Thanks for the PR!

For me to understand, the loop body already has some `LineNumberNode`s which are harmless (and may actually be useful for stacktraces if something within `@manipulate` errors), but the last one should be avoided, because it messes with the return value.

May be worth to also add a link to the PR to julia in the comments in the source code.

File: src/manipulate.jl (hunk 1 above)
Comment (simeonschaub) [reply src/manipulate.jl]: Well, that's embarassing... ^^ Yes, you are absolutely right!

File: src/manipulate.jl (hunk 2 above)
Comment (simeonschaub) [reply src/manipulate.jl]: Yeah, seems reasonable.

**Comment by simeonschaub** [comment]
> For me to understand, the loop body already has some LineNumberNodes which are harmless (and may actually be useful for stacktraces if something within @manipulate errors), but the last one should be avoided, because it messes with the return value.

Yes, blocks always return the return value of the last expression and if that's a `LineNumberNode`, that will evaluate to `nothing`.

> May be worth to also add a link to the PR to julia in the comments in the source code.

done.

**Review by piever — APPROVED** [review]

**Pull request merged**